src/Tasks/Microsoft.NET.Build.Tasks/CheckForDuplicateFrameworkReferences.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.Build.Framework;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    public class CheckForDuplicateFrameworkReferences : TaskBase
+    {
+        [Required]
+        public ITaskItem[] FrameworkReferenceItems { get; set; }
+
+        [Required]
+        public string MoreInformationLink { get; set; }
+
+        [Output]
+        public ITaskItem[] ItemsToAdd { get; set; }
+
+        [Output]
+        public ITaskItem[] ItemsToRemove { get; set; }
+
+
+        protected override void ExecuteCore()
+        {
+            var duplicateItems = FrameworkReferenceItems.GroupBy(i => i.ItemSpec, StringComparer.OrdinalIgnoreCase).Where(g => g.Count() > 1);
```

Comment:
Let's put the early return on know framework references here too for consistency. Can't hurt to spare the allocations.